wikiHandler.js

```diff
@@ -113,30 +129,36 @@ async function getWikiPricing(item_url, server) {
     //key for caching system is item_url underscore server
     const key = `${item_url}_${server}`
     //check cache before fetching
-    cache.get(key, (err, cached_data) => {
-        if (err) console.error(err);
-        //use cached data if available
-        if (cached_data !== null) {
-            return JSON.parse(cached_data);
-        } else {
-            //otherwise fetch new data
-            return fetch(item_url)
-                .then(response => response.text())
-                .then(text => {
-                    const priceData = parsePage(text, server);
-                    //arrays aren't valid redis keys
-                    const priceDataStr = JSON.stringify(priceData)
-                    
-                    //store parsed data in cache as a string
-                    cache.setex(key, cache_expiration, priceDataStr, (err) => {
-                        if (err) console.error(err);
+    return new Promise((resolve, reject) => {
```

Comment:
This was the offender.  I assumed redis functions would already be promisified but turns out they aren't.